src/utils/index.ts

```diff
@@ -0,0 +1,26 @@
+export type PartialDomRect = Partial<Omit<DOMRect, 'toJSON'>>;
+
+export const getBoundingContainerRect = (
+	element: EventTarget & HTMLDivElement,
+): PartialDomRect | null => {
+	if (!element) {
+		return null;
+	}
+	const parent = (element.offsetParent as HTMLElement) ?? document.body;
+	const top = element.offsetTop + parent.offsetTop;
+	const left = element.offsetLeft + parent.offsetLeft;
+	const height = element.clientHeight;
+	if (!parent || !top || !left || !height) {
+		return null;
+	}
+	return {
+		top,
+		bottom: top + height,
+		left,
+		right: left + element.clientWidth,
```

Comment:
Why do we need to check `element.clientHeight` but not `element.clientWidth`?

Reply:
nice catch